docs/benchmark.md

```diff
@@ -0,0 +1,73 @@
+# Benchmark
+
+Go benchmark tests that measure end-to-end performance of a running Ollama server. Run these tests to evaluate model inference performance on your hardware and measure the impact of code changes.
+
+## When to use
+
+Run these benchmarks when:
+- Making changes to the model inference engine
+- Modifying model loading/unloading logic
+- Changing prompt processing or token generation code
+- Implementing a new model architecture
+- Testing performance across different hardware setups
+
+## Prerequisites
+- Ollama server running locally (`127.0.0.1:11434`)
+- benchstat tool: `go install golang.org/x/perf/cmd/benchstat@latest`
+
+## Usage and Examples
+
+>[!NOTE]
+>All commands must be run from the `benchmark` directory of the Ollama project.
```

Comment:
Can you pass in `-bench=<dir>?

Reply:
Made this update to run from the root dir